One test has a problem in 32bit Python 2.7 with too large numbers (this is a
runtime problem, not pytest one). Other versions of Python are 64bit only,
and therefore this problem doesn't appear there.

Also, on Solaris it is not possible to remove cwd.

Several test_source.py tests are failing because they are not ready for 3.9.

Not suitable for upstream

--- pytest-4.4.0/testing/test_assertrewrite.py
+++ pytest-4.4.0/testing/test_assertrewrite.py
@@ -1233,6 +1233,8 @@ class TestIssue2121:
 
 
 @pytest.mark.parametrize("offset", [-1, +1])
+@pytest.mark.skipif(sys.version_info < (3, 0),
+    reason="32-bit Python 2.x cannot work with numbers this big.")
 def test_source_mtime_long_long(testdir, offset):
     """Support modification dates after 2038 in rewritten files (#4903).
 
@@ -1366,6 +1368,9 @@ class TestEarlyRewriteBailout(object):
     @pytest.mark.skipif(
         sys.platform.startswith("win32"), reason="cannot remove cwd on Windows"
     )
+    @pytest.mark.skipif(
+        sys.platform.startswith("sun"), reason="cannot remove cwd on Solaris"
+    )
     def test_cwd_changed(self, testdir, monkeypatch):
         # Setup conditions for py's fspath trying to import pathlib on py34
         # always (previously triggered via xdist only).
--- pytest-4.4.0/testing/code/test_source.py
+++ pytest-4.4.0/testing/code/test_source.py
@@ -127,6 +127,7 @@ def test_syntaxerror_rerepresentation():
     assert ex.value.text.strip(), "x x"
 
 
+@pytest.mark.skipif(sys.version_info >= (3, 9), reason="parser module is deprecated in 3.9")
 def test_isparseable():
     assert Source("hello").isparseable()
     assert Source("if 1:\n  pass").isparseable()
@@ -146,6 +147,7 @@ class TestAccesses(object):
     """
     )
 
+    @pytest.mark.skipif(sys.version_info >= (3, 9), reason="parser module is deprecated in 3.9")
     def test_getrange(self):
         x = self.source[0:2]
         assert x.isparseable()
@@ -478,6 +480,7 @@ def test_findsource():
     assert src[lineno] == "    def x():"
 
 
+@pytest.mark.skipif(sys.version_info >= (3, 9), reason="issues with 3.9")
 def test_getfslineno():
     from _pytest._code import getfslineno
 
